packages/block-editor/src/hooks/anchor.js

```diff
@@ -67,41 +68,56 @@ export const withInspectorControl = createHigherOrderComponent(
 			const hasAnchor = hasBlockSupport( props.name, 'anchor' );
 
 			if ( hasAnchor && props.isSelected ) {
+				const isWeb = Platform.OS === 'web';
+				const textControl = (
+					<TextControl
+						className="html-anchor-control"
+						label={ __( 'HTML anchor' ) }
+						help={
+							<>
+								{ __(
+									'Enter a word or two — without spaces — to make a unique web address just for this heading, called an “anchor.” Then, you’ll be able to link directly to this section of your page.'
+								) }
+
+								<ExternalLink
+									href={
+										'https://wordpress.org/support/article/page-jumps/'
+									}
+								>
+									{ __( 'Learn more about anchors' ) }
+								</ExternalLink>
+							</>
+						}
+						value={ props.attributes.anchor || '' }
+						valuePlaceholder={
+							! isWeb ? __( 'Add an anchor' ) : null
+						}
+						onChange={ ( nextValue ) => {
+							nextValue = nextValue.replace( ANCHOR_REGEX, '-' );
+							props.setAttributes( {
+								anchor: nextValue,
+							} );
+						} }
+						autoCapitalize="none"
+						autoComplete="off"
+					/>
+				);
+
 				return (
 					<>
 						<BlockEdit { ...props } />
-						<InspectorAdvancedControls>
-							<TextControl
-								className="html-anchor-control"
-								label={ __( 'HTML anchor' ) }
-								help={
-									<>
-										{ __(
-											'Enter a word or two — without spaces — to make a unique web address just for this heading, called an “anchor.” Then, you’ll be able to link directly to this section of your page.'
-										) }
-
-										<ExternalLink
-											href={
-												'https://wordpress.org/support/article/page-jumps/'
-											}
-										>
-											{ __( 'Learn more about anchors' ) }
-										</ExternalLink>
-									</>
-								}
-								value={ props.attributes.anchor || '' }
-								onChange={ ( nextValue ) => {
-									nextValue = nextValue.replace(
-										ANCHOR_REGEX,
-										'-'
-									);
-									props.setAttributes( {
-										anchor: nextValue,
-									} );
-								} }
-								autoComplete="off"
-							/>
-						</InspectorAdvancedControls>
+						{ isWeb && (
+							<InspectorAdvancedControls>
+								{ textControl }
+							</InspectorAdvancedControls>
+						) }
+						{ ! isWeb && props.name === 'core/heading' && (
```

Comment:
Is there a plan to remove the restriction to just the one block? 

I'd hope that as long as `TextControl` works well on mobile, it should be fairly straightforward to support this on all blocks that declare support for an HTML Anchor.

Reply:
Yes, I believe the plan is to implement other blocks in the future. For the time being, [Thomas and Stefanos requested](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1816#issuecomment-763199224) that we scope the work to Heading blocks.

From my testing, you are correct that wider anchor support is straightforward.

Reply:
> requested

Just a quick note on the process, as "requested" doesn't represent it very well: The way we work, at least on the native mobile apps, we make sure people collaborate by sharing information and context to design/implement solutions, including carving out the plan to get there.

We try to keep PR scope to the minimum so we can move land them in small but predictable steps (even if, say, landing means merging to a feature branch). We re-evaluate our plans as we go, when new info arises, like for example when we understand scope is actually too big for a single PR.

In our case here with the anchor feature, limiting to just the "Heading" block helps us pin down the implementation on mobile, add value by shipping the feature to the users, all while we making plans for the next steps.

Hope this helps, thanks!

Reply:
@hypest thank you for the feedback and context.

Reply:
Maybe we can add a comment here linking the issue where we plan to implement `InspectorAdvancedControls` and/or enable anchors for all the other blocks?